pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -333,6 +327,9 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) (Results, err
 
 func (s *Scheduler) trySchedule(ctx context.Context, p *corev1.Pod) error {
 	for {
+		if ctx.Err() != nil {
```

Comment:
We need to catch this and return at the call site now right, rather than emplacing it as a pod error?

Reply:
Good catch :)